### ISSUE TITLE
uBlock keeping show updating filters even if complete

### ISSUE THREAD

**Issue by 0x391F** [uBlockOrigin/uBlock-issues#3138]
### Prerequisites

- [X] I verified that this is not a filter list issue. Report any issues with filter lists or broken website functionality in the [uAssets issue tracker](https://github.com/uBlockOrigin/uAssets/issues).
- [X] This is **NOT** a [_YouTube_](https://www.github.com/uBlockOrigin/uAssets/issues/20586), [_Facebook_](https://www.github.com/uBlockOrigin/uAssets/issues/3367) or [_Twitch_](https://www.github.com/uBlockOrigin/uAssets/issues/5184) report. _These sites MUST be reported by clicking their **respective links**_.
- [X] This is not a support issue or a question. For support, questions, or help, visit [/r/uBlockOrigin](https://www.reddit.com/r/uBlockOrigin/).
- [X] I performed a [cursory search of the issue tracker](https://github.com/uBlockOrigin/uBlock-issues/issues?q=is%3Aissue) to avoid opening a duplicate issue.
- [X] The issue is not present after disabling uBO in the browser.
- [X] I checked the [documentation](https://github.com/gorhill/uBlock/wiki) to understand that the issue I am reporting is not normal behavior.

### I tried to reproduce the issue when...

- [X] uBO is the only extension.
- [X] uBO uses default lists and settings.
- [X] using a new, unmodified browser profile.

### Description

uBlock keeping show updating filters even if complete. 

### A specific URL where the issue occurs.

```yaml
N/A
```

### Steps to Reproduce

1. Update filters
2. Wait a few minutes
3. It still show updating, but if you refresh the page, it will show that the update has been complete (the updating indicator stop). 

### Expected behavior

It should stop the updating indicator. 

### Actual behavior

The updating indicator should auto stop after complete. 

### uBO version

1.56.0

### Browser name and version

Firefox 123.0 and latest Nightly (125.0a1)

### Operating System and version

Windows 10 LTSC 2021 x64 and Windows 11 VM

**Comment by 0x391F** [comment]
https://github.com/uBlockOrigin/uBlock-issues/assets/56181919/d7e74cd7-795d-4cb4-825b-5edbdf76dedb

**Comment by gwarser** [comment]
>"1. Update filters"

How? They update automatically just after you open the dashboard. What you did exactly?

**Comment by 0x391F** [comment]
> > "1. Update filters"
> 
> How? They update automatically just after you open the dashboard. What you did exactly?

For example, everyday I check uBlock Origin filters update manually because the filters update should complete before it run in sandboxie - I almost always run firefox in sandboxie.

**Comment by gorhill** [comment]
> I almost always run firefox in sandboxie.

Then possibly uBO is having trouble accessing its storage to cache the fetched data related to updating. To find out if this is a uBO-specific issue, you could try with other content blockers. If they also suffer similar issue, then the issue is specific to you and you will have to figure how to address it on your side.

Please look at the extension dev console to see if it says something:

- Go to `about:addons`
- Gear icon => "Debug Add-ons"
- Find uBO => Click "Inspect"
- Select "Console" pane

If something seems to be related to your issue, report it here.

**Comment by 0x391F** [comment]
I think this bug is appears since a version of uBlock, it used to auto stop loading, but it doesn't work after a certain day and/or uBlock Origin version.

**Comment by 0x391F** [comment]
I test on 1.54.0, this bug doesn't reproduce.

**Comment by 0x391F** [comment]
This bug maybe caused by that uBlock Origin won't update the indicator status even if filters update have been complete.

**Comment by gorhill** [comment]
> maybe caused

There is no point speculating, if we can't reproduce it's because there is something specific on your side triggering the issue. The goal is to identify what is that specific trigger on your side. This is why I asked:

> Please look at the extension dev console

Also, investigate whether sandboxie is that trigger.

**Comment by 0x391F** [comment]
> Also, investigate whether sandboxie is that trigger.

I can confirm it is not sandboxie trigger this bug, in [https://github.com/uBlockOrigin/uBlock-issues/issues/3138#issuecomment-1962793859](https://github.com/uBlockOrigin/uBlock-issues/issues/3138#issuecomment-1962793859), this bug also happens on Windows 11 vm without runing in sandboxie.

**Comment by gorhill** [comment]
> Please look at the extension dev console

Also, provide troubleshooting information from Support pane.

**Comment by 0x391F** [comment]
> Please look at the extension dev console to see if it says something:
> 
>     * Go to `about:addons`
> 
>     * Gear icon => "Debug Add-ons"
> 
>     * Find uBO => Click "Inspect"
> 
>     * Select "Console" pane
> 
> 
> If something seems to be related to your issue, report it here.

Here's no items in console. 
![1](https://github.com/uBlockOrigin/uBlock-issues/assets/56181919/0fe1147d-6bec-4852-b754-ea94964d6c82)

**Comment by 0x391F** [comment]
uBlock Origin: 1.56.0
Firefox: 123
filterset (summary):
 network: 354802
 cosmetic: 103182
 scriptlet: 36718
 html: 1786
listset (total-discarded, last-updated):
 added:
  adguard-generic: 79275-5381, 6h.46m
  adguard-mobile: 9090-102, 6h.46m
  adguard-spyware-url: 1467-106, 6h.46m
  adguard-spyware: 113905-33606, 6h.46m
  block-lan: 56-0, 21d.8h.28m
  curben-phishing: 104549-39, 7m
  dpollock-0: 11649-492, 3d.6h.48m
 default:
  user-filters: 0-0, never
  ublock-filters: 37633-146, 9m Δ
  ublock-badware: 8009-14, 9m Δ
  ublock-privacy: 741-0, 9m Δ
  ublock-unbreak: 2280-1, 9m Δ
  ublock-quick-fixes: 106-1, 9m Δ
  easylist: 81757-863, 9m Δ
  easyprivacy: 50285-990, 9m Δ
  urlhaus-1: 8770-0, 7m
  plowe-0: 3782-1, 4d.6h.56m
  CHN-0: 25354-156, 1d.6h.18m
filterset (user): [empty]
userSettings:
 prefetchingDisabled: false
hiddenSettings: [none]
supportStats:
 allReadyAfter: 1007 ms
 maxAssetCacheWait: 535 ms
 cacheBackend: indexedDB

**Comment by gorhill** [comment]
> Can you record video from the start (before you open the browser)?

It's not clear how you force an update, your video just just the issue itself, not what was done to make the issue appear.

**Comment by Iz-zzzzz** [comment]
I also encounter this issue occasionally, but I'm unable to reproduce it consistently.

**Comment by 0x391F** [comment]
This bug also happens on a new-install Fenix (Firefox for Android) 123.0 with uBlock Origin 1.56.0.

**Comment by Iz-zzzzz** [comment]
It seems that this issue may have been caused by repeatedly restoring uBO settings from an older version. I resolved it by resetting uBO and configuring everything from scratch.

**Comment by 0x391F** [comment]
This bug doesn't happens on uBlock Origin 1.56.0 with Edge 122.0.2365.92.

**Comment by gwarser** [comment]
This always happen to all lists? Not only few which have been outdated at the time?

**Comment by 0x391F** [comment]
> Not only few which have been outdated at the time?
Yes, it only happens on out-of-date lists.

**Comment by garry-ut99** [comment]
Video from https://github.com/uBlockOrigin/uBlock-issues/issues/3138#issuecomment-1962793859 is broken for me, on all browsers (Firefox , Opera , Edge), when Play is clicked, seek bar jumps instantly to the end, I don't know whether the video has been removed, or is private or is broken, nor I know why.

> Yes, it only happens on out-of-date lists.

On random out-of-date lists or maybe they are always the same lists (some specific lists cause it)?

**Comment by 0x391F** [comment]
> Video from [#3138 (comment)](https://github.com/uBlockOrigin/uBlock-issues/issues/3138#issuecomment-1962793859) is broken for me, on all browsers (Firefox , Opera , Edge), when Play is clicked, seek bar jumps instantly to the end, I don't know whether the video has been removed, or is private or is broken, nor I know why.

For me, it plays without any problem on Firefox and Edge, maybe there's something wrong with your network.

**Comment by garry-ut99** [comment]
For me not, I still use Win 7 SP1, hence I use Firefox 115 ESR + Opera 95 + Edge 109, and the video is fu..... broken in all of them, I've tried now on Firefox 124 Mobile (which I forgot to check) and video plays fine here.
It reminds me the previous issue from 2 days ago, which turned out to happen on FF 115 ESR only :
`https://github.com/uBlockOrigin/uBlock-issues/issues/3161#issuecomment-2013223395`
It's getting worse by time: more and more things broken week by week in obsolete browsers, I've now replaced Opera & Edge with https://github.com/win32ss/supermium as it's based on Chromium 122 and not on 109, and video works fine here as well.

</details>

<hr>

Getting back to the issue : would be good to know what says uBO logger set on "All tabs", when the issue occurs, maybe can determine whether it's related to some specific filter lists / caused by some server lag/delays or not.

**Comment by 0x391F** [comment]
> What I have observed is that this bug shows on a clean install of uBlock on a clean new Firefox profile with the latest main release version of Firefox on either Windows or macOS, but only if the filter lists settings are opened in private windows.
> 
> This bug does not happen if you open the settings and update the filter lists in a non-private window. There, all the animations finish correctly when the lists are successfully updated.

Yes, you're right.

**Comment by gorhill** [comment]
After a quick investigation, it appears BroadcastChannel does not work to send messages from non-private to private contexts. The result is that when the dashboard is opened in a private window, it never receives the broadcast channel messages from the non-private background process of uBO.

**Comment by przemo20** [comment]
I ran into a similar problem (the loading icon animation doesn't disappear, even though the filters got correctly updated).
_(Not sure if I'm allowed to share this here, since this is related, or should I start a new issue/bug)_
To reproduce the issue, you will need to have Firefox, then:

1. Install uBlock Origin
2. Allow uBO extension to run in a private window
3. Open a new **private window**
4. Click on uBO extension icon, click on the gear icon in its popup menu (to open the extension settings page)
5. Go to the filter list tab, then click to manually force update of one of the filters

The spinning animation keeps playing, but if you go to about:debugging, inspect uBO, switch to the Network tab, and try to update a different filter list, you will see that the request to fetch list update was successful (status 200).

Updating filters in the normal FF window, works correctly.
Chrome's private window doesn't seem to be affected.

This issue also happens on the “Report a filter issue” page (that shows up after clicking on the “Report an issue on this website” button) after clicking on the “Update now” button there – the “Verify that the issue still exists after reloading the problematic webpage.” message never appears.

Here's my troubleshooting information:
```
uBlock Origin: 1.59.0
Firefox: 130
filterset (summary):
 network: 156597
 cosmetic: 100179
 scriptlet: 28475
 html: 2268
listset (total-discarded, last-updated):
 added:
  adguard-popup-overlays: 28019-216, 30m
  adguard-mobile-app-banners: 5482-25, 30m
  adguard-other-annoyances: 14555-296, 30m
  adguard-widgets: 2666-13, 30m
  ublock-annoyances: 5915-167, 30m
 default:
  user-filters: 78-0, never
  ublock-filters: 40595-138, 9m
  ublock-badware: 11169-6, 9m
  ublock-privacy: 1248-22, 9m
  ublock-unbreak: 2538-1, 9m
  ublock-quick-fixes: 134-4, 9m
  easylist: 85562-186, 30m
  easyprivacy: 53092-62, 30m Δ
  urlhaus-1: 25255-0, 30m
  plowe-0: 3546-993, 30m
  POL-0: 8764-49, 30m
  POL-2: 1283-51, 30m
filterset (user): [array of 79 redacted]
trustedset:
 added: [array of 36 redacted]
switchRuleset:
 added: [array of 1 redacted]
userSettings:
 largeMediaSize: 2000
 showIconBadge: false
hiddenSettings: [none]
supportStats:
 allReadyAfter: 846 ms (selfie)
 maxAssetCacheWait: 234 ms
 cacheBackend: indexedDB
```
